server/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```diff
@@ -199,7 +200,12 @@ private static Object convertValueFromSortType(String fieldName, SortField.Type
 
                 case STRING_VAL:
                 case STRING:
```

Comment:
Oh boy. This is a funny case where `STRING` actually means "sort by a byte array. Oh yeah, and usually that's utf-8 encoded string, but not always". Yikes. Sad Nik.

Reply:
But I don't think there is much you can do about it because its a lucene concept.

Reply:
@romseygeek may be able to double check on this. I think it's right, but still sad.

Reply:
BytesRefs compare byte-by-byte, no need to worry about UTF-8 encoding or whatever.

Reply:
TSID is stored as a SortedSetDocValuesField, so what will *actually* get compared here is their ordinal number - SSDV is stored as a terms dictionary containing all the (sorted and deduplicated) values, and then a per-document ordinal.  So to retrieve the value you get the ord for the current doc and then get the value from the terms dictionary, but to sort on it you can just compare their ordinal values - much faster than comparing byte by byte.

Reply:
> what will _actually_ get compared here is their ordinal number

Yeah. I should have said "sort as though by byte array" - it's using ordinals. The thing that bothers me is that it uses the word `STRING` here when it means `BYTES`. But I'm being grumbly.